docs/recipes/recipe-data-structures-in-git/index.md

```diff
@@ -360,7 +360,206 @@ Validation has passed. Now our colleagues can feedback on our changes and if eve
+:::
+
+:::caution Warning
+
+The `iglu:com.example/login/jsonschema/1-0-1` data structure has to be deployed at least to a develop envinroment. Currently referencing local data structures is not supported
+
+:::
+
+We can run the same `publish --dry-run` command as before, to see if the output is as expected. The output should contain the following lines
+
+```bash
+snowplow-cli dp publish --dry-run
+```
+
+```
+INFO publish will create data product file=.../data-products/login.yaml name=Login resource name=0edb4b95-3308-40c4-b266-eae2910d5d2a
+INFO publish will update event specifications file=.../data-products/login.yaml name="Login success" resource name=cfb3a227-0482-4ea9-8b0d-f5a569e5d103 in data product=0edb4b95-3308-40c4-b266-eae29
+```
+
+We can apply the changes by using the publish command without the `--dry-run` flag
+
+```bash
+snowplow-cli dp publish
+```
+
+### Add data products validation and publishing in the github actions
+
+Now that we've modeled a source app, data product and event spec, let's see how we can add them to the existing github actions workflows for data structures. You can customize your setup, use a separate repository or a separate actions, but in this example we'll add the data products publishing into the existing workflows.
+
+Lets modify the PR example, and add the following line. This command will validate and print the changes to the github actions log.
+
+```yml {20} title=".github/workflows/validate-pull-request.yml"
+on:
+  pull_request:
+    branches: [develop, main]
+
+jobs:
+  validate:
+    runs-on: ubuntu-latest
+    env:
+      SNOWPLOW_CONSOLE_ORG_ID: ${{ secrets.SNOWPLOW_CONSOLE_ORG_ID }}
+      SNOWPLOW_CONSOLE_API_KEY_ID: ${{ secrets.SNOWPLOW_CONSOLE_API_KEY_ID }}
+      SNOWPLOW_CONSOLE_API_KEY: ${{ secrets.SNOWPLOW_CONSOLE_API_KEY }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: snowplow-product/setup-snowplow-cli@v1
+
+      - run: snowplow-cli ds validate --gh-annotate
+
+      - run: snowplow-cli dp publish --dry-run --gh-annotate
+```
+
+Data products, source apps and event specs don't have the dev and prod environments, so it's enough to publish them once.
+We can add the same command but without the `--dry-run` flag to the publish pipeline.
+
+```yml {20} title=".github/workflows/publish-develop.yml"
+on:
+  push:
+    branches: [develop]
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    env:
+      SNOWPLOW_CONSOLE_ORG_ID: ${{ secrets.SNOWPLOW_CONSOLE_ORG_ID }}
+      SNOWPLOW_CONSOLE_API_KEY_ID: ${{ secrets.SNOWPLOW_CONSOLE_API_KEY_ID }}
+      SNOWPLOW_CONSOLE_API_KEY: ${{ secrets.SNOWPLOW_CONSOLE_API_KEY }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: snowplow-product/setup-snowplow-cli@v1
+
+      - run: snowplow-cli ds publish dev --managed-from $GITHUB_REPOSITORY
+
+      - run: snowplow-cli dp publish --gh-annotate
```

Comment:
Is `--gh-annotate` needed here?

Reply:
Nope, noting to annotate on a merged PR, good catch